server.ts

```diff
@@ -80,28 +85,31 @@ const getsurvey = async (query: string | ParsedQs, req: Request<{}>, res: Respon
 app.get("/s/", async (req, res) => {
   //For debugging purposes. Prints encrypted version of url. In the future, this will be done in researcher menu.
   const cipher = crypto.createCipheriv(crypto_algorithm, private_key_example, iv_example);
-  let encrypted = cipher.update(JSON.stringify(req.query), 'utf8', 'hex');
-  console.log(encrypted += cipher.final('hex'));
-
+  let encrypted = cipher.update(JSON.stringify(req.query), "utf8", "hex");
+  console.log(encrypted += cipher.final("hex"));
   console.log(req.query);
   getsurvey(req.query, req, res);
 });
 
-app.get('/e/:data', async (req, res) => {
+app.get("/e/:data", async (req, res) => {
   //in the future, private_key and iv will be obtained through researcher database
   try {
     const decipher = crypto.createDecipheriv(crypto_algorithm, private_key_example, iv_example);
-    let decrypted = decipher.update(req.params.data, 'hex', 'utf8');
-    decrypted += decipher.final('utf8');
+    let decrypted = decipher.update(req.params.data, "hex", "utf8");
+    decrypted += decipher.final("utf8");
     getsurvey(JSON.parse(decrypted), req, res);
   } catch (error) {
     console.error(error);
-    res.redirect('/');
+    res.redirect("/");
   }
 });
 
-app.post("/survey", (req, res) => {
+app.post("/survey", csrfProtection, (req, res) => {
   responses.insert(req.body);
```

Comment:
This function stores the data to DB, so probably we want to do it after adding the time variables to `req.body`.

Reply:
Reordered in latest commit